### ISSUE TITLE
Possibility to return pure text from render() without wrapping it in an element

### ISSUE THREAD

**Issue by xkr47** [facebook/react#7353]
I'm asking if, in the light of #5753, it would be possible to also return plain text from `render()` as an alternative to the current `ReactElement` return type such that one would not have to wrap it in an element. Could the same comment-wrapping method allow for this also for render methods?

``` js
// Before
return <span>{this.props.text}</span>;
// After
return this.props.text;
```

Motivation is that it is sometimes hard to get layout to work as expected if one has to introduce span elements, and the alternative is to write plain functions instead of react components to produce the expected result. It would be syntactically nice to be able to componentize text-only components the same way as "html-enabled" components.

**Comment by stephenkingsley** [comment]
I think this is convenient. and the code look like simplify.

**Comment by syranide** [comment]
@xkr47 Not disagreeing. But be mindful of overusing text-only components, components have non-trivial overhead and could have significant performance impact if overused.

**Comment by xkr47** [comment]
@syranide true.. are there any benchmark results on these kinds of things? inline vs function call vs stateless component vs stateful component..?

**Comment by steida** [comment]
I don't think it's a good idea because it goes against how React Native with inline styles works. Tie your hands to free your mind. When every text must be wrapped in Text component, we can style it easily without global styles.

**Comment by gaearon** [comment]
FWIW this is already implemented in Fiber (#6170).
We won't be bringing this to the old reconciler.

**Comment by aweary** [comment]
Closing this since it's already implemented in the next major and it won't be added to any 15.5 release
